tests/integration/aof-multi-part.tcl

```diff
@@ -0,0 +1,1065 @@
+            assert_equal v2 [$client get k2]
+            assert_equal v3 [$client get k3]
+            
+            assert_equal 0 [check_file_exist $server_path $aof_basename]
+            assert_equal 1 [check_file_exist $aof_dirpath $aof_basename]
+           
+            assert_aof_manifest_content $aof_manifest_filepath  {
+                {file appendonly.aof seq 1 type b} 
+                {file appendonly.aof.1.incr.aof seq 1 type i}
+            }
+        }
+    }
+
+    clean_aof_persistence $aof_dirpath
+
+    # Tests18: Multi Part AOF can be loaded correctly when both server dir and aof dir contain old AOF
+    create_aof $aof_old_version_filepath {
+        append_to_aof [formatCommand set k1 v1]
+        append_to_aof [formatCommand set k2 v2]
+        append_to_aof [formatCommand set k3 v3]
+    }
+
+    create_aof_dir $aof_dirpath
+    create_aof_manifest $aof_manifest_filepath {
+        append_to_manifest "file appendonly.aof seq 1 type b\n"
+    }
+
+    create_aof $aof_old_version_newpath {
+        append_to_aof [formatCommand set k4 v4]
+        append_to_aof [formatCommand set k5 v5]
+        append_to_aof [formatCommand set k6 v6]
+    }
+
+    start_server_aof [list dir $server_path] {
+        test "Multi Part AOF can be loaded correctly when both server dir and aof dir contain old AOF" {
+            assert_equal 1 [is_alive $srv]
+
+            set client [redis [dict get $srv host] [dict get $srv port] 0 $::tls]
+            wait_done_loading $client
+
+            assert_equal 0 [$client exists k1]
+            assert_equal 0 [$client exists k2]
+            assert_equal 0 [$client exists k3]
+
+            assert_equal v4 [$client get k4]
+            assert_equal v5 [$client get k5]
+            assert_equal v6 [$client get k6]
+            
+            assert_equal 1 [check_file_exist $server_path $aof_basename]
+            assert_equal 1 [check_file_exist $aof_dirpath $aof_basename]
+           
+            assert_aof_manifest_content $aof_manifest_filepath  {
+                {file appendonly.aof seq 1 type b} 
+                {file appendonly.aof.1.incr.aof seq 1 type i}
+            }
+        }
+    }
+
+    clean_aof_persistence $aof_dirpath
+    catch {exec rm -rf $aof_old_version_filepath}
+
+    # Tests19: Multi Part AOF can't load data when the manifest contains the old AOF file name but the file does not exist in server dir and aof dir
+    create_aof_dir $aof_dirpath
+    create_aof_manifest $aof_manifest_filepath {
+        append_to_manifest "file appendonly.aof seq 1 type b\n"
+    }
+    start_server_aof [list dir $server_path] {
+        test "Multi Part AOF can't load data when the manifest contains the old AOF file name but the file does not exist in server dir and aof dir" {
+            wait_for_condition 100 50 {
+                ! [is_alive $srv]
+            } else {
+                fail "AOF loading didn't fail"
+            }
+        }
```

Comment:
let's add an assertion for some log message here too (all tests that expects the server to fail starting)

Reply:
Added